elyra/pipeline/kfp/component_parser_kfp.py

```diff
@@ -63,40 +63,60 @@ def _parse_properties(self, component_yaml: Dict[str, Any]) -> List[ComponentPar
-                                                            io_object_type="input",
-                                                            parameter_type=data_type)
-
-            data_type, control_id, default_value = self.determine_type_information(data_type)
-
-            # Get value if provided
-            value = param.get('default', '')
-
-            ref = param.get('name').lower().replace(' ', '_')
-            properties.append(ComponentParameter(id=ref,
-                                                 name=param.get('name'),
-                                                 data_type=data_type,
-                                                 value=(value or default_value),
-                                                 description=description,
-                                                 control_id=control_id,
-                                                 required=required))
+        output_params = component_yaml.get('outputs', [])
+
+        all_params = {"inputs": input_params, "outputs": output_params}
+
+        # Loop through inputs and outputs and create custom properties
+        for param_type, params in all_params.items():
+            for param in params:
+                # KFP components default to being required unless otherwise stated.
+                # Reference: https://www.kubeflow.org/docs/components/pipelines/reference/component-spec/#interface
+                required = True
+                if param.get('optional') is True:
+                    required = False
+
+                # Assign type, default to string
+                data_type = param.get('type', 'string')
+
+                # Set description and include parsed type information
+                description = self._format_description(description=param.get('description', ''),
+                                                       data_type=data_type)
+
+                # Change type to reflect the parameter type (inputValue vs inputPath vs outputPath)
+                if self._is_path_based_parameter(param.get('name'), component_yaml):
+                    data_type = f"{param_type[:-1]}Path"
+
+                data_type_info = self.determine_type_information(data_type)
+                if data_type_info.undetermined:
+                    self.log.warning(f"Data type from parsed data ('{data_type}') could not be determined. "
+                                     f"Proceeding as if 'string' was detected.")
+
+                if not data_type_info.required:
+                    required = data_type_info.required
+
+                # Get value if provided
+                value = param.get('default', '')
+
+                ref_name = param.get('name').lower().replace(' ', '_')
+                name = param.get('name')
+
+                if data_type_info.data_type == 'outputpath':
+                    ref_name = f"output_{ref_name}"
+                    description = f"{description}. This is a(n) output of this component."
```

Comment:
Not sure if there's a case it'd be `a` instead of `an`, hence

```suggestion
                    description = f"{description}. This is an output of this component."
```